pkg/minikube/kubeconfig/config.go

```diff
@@ -178,3 +182,58 @@ func decode(data []byte) (*api.Config, error) {
 
 	return config.(*api.Config), nil
 }
+
+// GetKubeConfigStatus verifys the ip stored in kubeconfig.
+// If incorrect, the ip is overwritten.
+func GetKubeConfigStatus(ip string, filename string) (string, error) {
+	if ip == "" {
+		return "", fmt.Errorf("Error, empty ip passed")
+	}
+	ks := ""
+	kip := ""
+	con, err := ReadConfigOrNew(filename)
+	if err != nil {
+		return "", errors.Wrap(err, "Error getting kubeconfig status")
+	}
+	if con.Clusters["minikube"] == nil {
+		return "", errors.Wrap(err, "Kubeconfig does not have a record of a minikube cluster")
+	}
+	kip = con.Clusters["minikube"].Server
+
+	if strings.Contains(kip, ip) {
+		ks = "Correctly configured"
+	} else {
+		ks = "Kubeconfig IP incorrectly configured.  Run minikube update-context to correct."
+	}
+	return ks, nil
+}
+
+// UpdateKubeconfigIP overwrites the IP stored in kubeconfig with the provided IP.
+func UpdateKubeconfigIP(ip string, filename string) (string, error) {
```

Comment:
Yes, this is called by the other functionality Aaron suggested, the `minikube update-context` command line call.  The file for that functionality hasn't been pushed yet, should be in Monday.